intel-lustre-clients-on-centos/azuredeploy.json

```diff
@@ -0,0 +1,403 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "artifactsBaseUrl": {
+            "type": "string",
+            "defaultValue":"https://raw.githubusercontent.com/arsenvlad/azure-quickstart-templates/master/intel-lustre-clients-on-centos",
+            "metadata": {
+                "description": "Base URL of the solution template gallery package",
+                "artifactsBaseUrl": ""
+            }
+        },
+        "location": {
+            "type": "string",
+            "allowedValues": [
+                "westus",
+                "eastus2",
+                "centralus",
+                "southcentralus",
+                "northcentralus",
+                "westeurope",
+                "northeurope",
+                "eastasia",
+                "southeastasia",
+                "japanwest",
+                "japaneast",
+                "brazilsouth",
+                "australiaeast",
+                "australiasoutheast",
+                "centralindia",
+                "westindia",
+                "southindia",
+                "centralindia"
+            ],
+            "metadata": {
+                "description": "Location for the Lustre nodes"
+            }
+        },
+        "imageSku": {
+            "type": "string",
+            "defaultValue": "6.6",
+            "allowedValues": [
+                "6.6",
+                "7.0"
+            ],
+            "metadata": {
+                "description": "OpenLogic CentOS version to use"
+            }
+        },
+        "adminUsername": {
+            "type": "string",
+            "defaultValue": "azureuser",
+            "metadata": {
+                "description": "Admin username for the virtual machines"
+            }
+        },
+		"authenticationType": {
+            "type": "string",
+            "defaultValue": "password",
+            "allowedValues": [
+                "password",
+                "sshPublicKey"
+            ],
+            "metadata": {
+                "description": "Authentication type for the virtual machines"
+            }
+        },
+        "adminPassword": {
+            "type": "securestring",
+			"defaultValue": "",
```

Comment:
The default value of "empty" is necessary since this template supports either password or sshPublicKey. Therefore, the required field depends on the selection of the Authentication Type parameter. When "password" is selected adminPassword is required, when "sshPublicKey" is selected "sshPublicKey" is required and password should be empty. If defaultValue:"" is removed data needs to be entered into both fields. The failure with both fields being empty will be quick - platform will reject empty password before deploying the VM.